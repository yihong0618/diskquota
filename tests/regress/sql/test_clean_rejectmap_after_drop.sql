CREATE DATABASE test_clean_rejectmap_after_drop;

\c test_clean_rejectmap_after_drop
CREATE EXTENSION diskquota;

\! gpconfig -c "diskquota.hard_limit" -v "on" > /dev/null
\! gpstop -u > /dev/null

CREATE ROLE r;
SELECT diskquota.set_role_quota('r', '1MB');
CREATE TABLE b (t TEXT) DISTRIBUTED BY (t);
ALTER TABLE b OWNER TO r;
SELECT diskquota.wait_for_worker_new_epoch();

INSERT INTO b SELECT generate_series(1, 100000000); -- fail

SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;

INSERT INTO b SELECT generate_series(1, 100); -- ok

\c contrib_regression
DROP DATABASE test_clean_rejectmap_after_drop;
DROP ROLE r;

\! gpconfig -c "diskquota.hard_limit" -v "off" > /dev/null
\! gpstop -u > /dev/null
