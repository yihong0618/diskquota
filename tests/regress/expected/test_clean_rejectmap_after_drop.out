CREATE DATABASE test_clean_rejectmap_after_drop;
\c test_clean_rejectmap_after_drop
CREATE EXTENSION diskquota;
\! gpconfig -c "diskquota.hard_limit" -v "on" > /dev/null
\! gpstop -u > /dev/null
CREATE ROLE r;
NOTICE:  resource queue required -- using default resource queue "pg_default"
SELECT diskquota.set_role_quota('r', '1MB');
 set_role_quota 
----------------
 
(1 row)

CREATE TABLE b (t TEXT) DISTRIBUTED BY (t);
ALTER TABLE b OWNER TO r;
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

INSERT INTO b SELECT generate_series(1, 100000000); -- fail
ERROR:  role's disk space quota exceeded with name: 34523  (seg0 127.0.0.1:6002 pid=23690)
SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
INSERT INTO b SELECT generate_series(1, 100); -- ok
\c contrib_regression
DROP DATABASE test_clean_rejectmap_after_drop;
DROP ROLE r;
\! gpconfig -c "diskquota.hard_limit" -v "off" > /dev/null
\! gpstop -u > /dev/null
